### DISCUSSION TITLE
Discussion about graph autoencoding architectures

### DISCUSSION THREAD

**Discussion by hkim716** [pyg-team/pytorch_geometric#4412] [Q&A] [answered]
Hi Matt, 
I'm trying to design a graph autoencoder that reconstructs node features. 
[This paper](https://link.springer.com/chapter/10.1007/978-3-030-93722-5_11) showed one of my interests because the model predicts wall shear stresses, which are the node features. 
In figure 2, I think they used pre-scheduled coarsened graphs for down (encoder) and up (decoder) scaling process. 

Is it necessary to store ```pre-defined coarsened graphs``` to decode/reconstruct the original graph?
For instance, if we use ```graclus``` without storing any intermediate clustering information, is there a way that we could reconstruct the original graph?

**Comment by rusty1s** [comment, marked as answer]
Yeah, most graph autoencoders assume that downsampled and upsampled graph structure is additionally given, which somewhat eliminates the need to decode node features in the first place. Reconstructing intermediate and final graph representations is still a challenging problem (especially for meshes which have additional constraints on how the graph structure should look like). At the top of my mind, I cannot think of any paper that tries to tackle this yet.
